frontend/hooks/useBalanceContext.ts

```diff
@@ -52,3 +58,54 @@ export const useServiceBalances = (serviceConfigId: string) => {
     isLowBalance,
   };
 };
+
+/**
+ * Balances relevant to the master wallets, eoa, and safes
+ */
+// TODO: complete this hook
+export const useMasterBalances = () => {
+  const { masterSafes, masterEoa } = useMasterWalletContext();
+  const { walletBalances, lowBalances, stakedBalances } = useBalanceContext();
+
+  const masterWalletBalances = useMemo(
+    () =>
+      walletBalances?.filter((balance) =>
+        flatAddresses.includes(balance.walletAddress),
+      ),
+    [flatAddresses, walletBalances],
+  );
+
+  const masterStakedBalances = useMemo(
+    () =>
+      stakedBalances?.filter((balance) =>
+        flatAddresses.includes(balance.walletAddress),
+      ),
+    [flatAddresses, stakedBalances],
+  );
+
+  const masterLowBalances = useMemo(
+    () => lowBalances?.filter((balance) => balance.walletAddress),
+    [lowBalances],
+  );
+
+  const isLowBalance = useMemo(
+    () => masterLowBalances?.length > 0,
+    [masterLowBalances],
+  );
+
+  const masterSafeBalances = useMemo<WalletBalanceResult[]>(
+    () =>
+      walletBalances?.filter((balance) =>
+        masterSafes.find(({address}) => balance.walletAddress === address),
```

Comment:
I see the TODO, so can ignore, just a thought - does it make sense to use flatAddresses here for consistency?

Reply:
yeah, for sure, will update and add more consistently here